### DISCUSSION TITLE
How to create a raw link from Github

### DISCUSSION THREAD

**Discussion by imtiazshawn** [community/community#44370] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I want to get a raw link like this "https://raw.githubusercontent.com/vsnaichuk/vsnaichuk/master/CV.pdf".
I upload my resume to a github repository. but I don't get any raw option. how can I get it?

**Comment by surjendu104** [comment, marked as answer]
You can easily create this by filling the link with your details.........
https://raw.githubusercontent.com/<your_Github_username>/<your_repository_name>/<branch_name>/<file_name>.<extension_name>
Aletrnatively, 
https://github.com/<your_Github_username>/<your_repository_name>/<branch_name>/<file_name>.<extension_name>

try this one. If it does not work then sent me the repo link

**Reply by zulkefal** [reply]
hello its not wokring for me
https://github.com/zulkefal/resume/blob/main/Zulkefal_Resume%20(3).pdf

**Reply by surjendu104** [reply]
I am not able to see the a repository named `resume` in your git profile. Is it private repo.
If it is private, it is not publicly accessible.

**Reply by Tuscan-blue** [reply]
> Github uses `content-disposition: attachment` for PDFs, which downloads the file automatically on github.com. You can host the file on a static page provided by GitHub which can link to this file and update as you update the PDF. For that, you can see [github pages](http://pages.github.com/).

resources: https://webapps.stackexchange.com/questions/48061/can-i-trick-github-into-displaying-the-pdf-in-the-browser-instead-of-downloading

**Reply by RajarshiMaiti** [reply]
I tried this method but still couldn't do it. My repository is public but still couldn't generate the Raw url

**Reply by MappleGaming208** [reply]
use ?raw=true so for ex i wanna make "https://github.com/MappleGaming208/Magic/blob/main/Scripts/lua/load.lua" into raw just do "https://github.com/MappleGaming208/Magic/blob/main/Scripts/lua/load.lua?raw=true"
it will redirect to "https://raw.githubusercontent.com/MappleGaming208/Magic/refs/heads/main/Scripts/lua/load.lua" youre welcome

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @imtiazshawn! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!

**Comment by oluwatosinjoseph** [comment]
Actually, I am trying to run a node and I'm stuck here because I can't create a raw link.

**Comment by Tuscan-blue** [comment]
Maybe this could help you.
![image](https://github.com/community/community/assets/141131070/7fac7500-50d9-4f3e-ab2f-e26d2f587409)

**Reply by DewanjaliKapse** [reply]
Its not working for me. Instead downloading the csv file.

**Reply by Tuscan-blue** [reply]
resources: https://webcache.googleusercontent.com/search?q=cache:https://medium.com/geekculture/how-to-read-csv-files-from-github-com-directly-ea9de687653d

**Reply by BensonCoolfi** [reply]
@DewanjaliKapse 
> Its not working for me. Instead downloading the csv file.

you need to right-cilck it and choose Copy link address
![image](https://github.com/user-attachments/assets/65ae9943-02f6-4119-8be5-627883655af1)

**Reply by haritha101513** [reply]
Hey, 
Thanks Benson. Actually I was also having the same issue and worked with your solution.

**Reply by lssiWOW** [reply]
Hi,
@Tuscan-blue this actually works bro THANKS

**Comment by hendrotirtaanggardi** [comment]
> Mungkin ini bisa membantu Anda. ![gambar](https://private-user-images.githubusercontent.com/141131070/316443640-7fac7500-50d9-4f3e-ab2f-e26d2f587409.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE0MzcxOTcsIm5iZiI6MTcxMTQzNjg5NywicGF0aCI6Ii8xNDExMzEwNzAvMzE2NDQzNjQwLTdmYWM3NTAwLTUwZDktNGYzZS1hYjJmLWUyNmQyZjU4NzQwOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNlQwNzA4MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZDc0MmM2MDFhZjQ2MzU1NWQ4Njk4Mzg2MGIxMWIzNTRiZjYxN2Y3ZmU0ZTkyZGI2M2Q2Y2ZkMDg4YjZiYzE4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0dpnKn-MPCG2j7OmWjL7No9SSh4wmkIA6XaASKzUmOk)

Halo kak, dimana ada button raw selain di gambar ini? di tampilan github saya tidak muncul

**Reply by Tuscan-blue** [reply]
1. Go to your Git repository where the file is located.
2. Find the file you want to get the raw content link for.
3. Click on the file to open it.
4. In the file view, look for the "Raw" button or link. This is usually located near the top of the file view.

For example: https://github.com/hendrotirtaanggardi/hendrotrtanggardi/blob/main/index.html
You can find the "Raw" button in this link.

**Reply by codershreya2704** [reply]
How can i make an image as a raw link?

**Reply by Tuscan-blue** [reply]
> How can i make an image as a raw link?

Perhaps what you need is an image hosting service.

**Reply by enrico3** [reply]
> How can i make an image as a raw link?

This worked for me: Navigate to the image on Github similar to the screenshots above so that you see the image embedded in the Github page. Now right click on the image itself and select 'open image in new tab'. In the new browser tab I now have the raw URL in the address bar.

**Comment by Hawk983** [comment]
GitHub is down

**Comment by Kickwinnerz** [comment]
![IMG-20241214-WA0009](https://github.com/user-attachments/assets/47742a1e-1175-4e02-8726-587b2d289e6e)
[](url)

**Comment by Kickwinnerz** [comment]
![IMG-20241214-WA0009](https://github.com/user-attachments/assets/47742a1e-1175-4e02-8726-587b2d289e6e)
[](url)

**Comment by deleonroque75-cyber** [comment]
Oo boss 😎 pwede tayong gumawa ng GitHub repo para sa script mo para mas madali mong ma-load gamit:

loadstring(game:HttpGet("https://raw.githubusercontent.com/USERNAME/REPONAME/main/filename.lua"))()

⚠️ Pero sabi mo bawal sa’yo yung loadstring, so mas ok gagawin natin:

Gagawa ako ng GitHub Gist/Repo link

Diretso kang makaka-copy/paste ng buong script doon (no need loadstring)



---

❓ Tanong boss:
Gusto mo ba na ako na mismo gumawa ng repo tapos ibigay ko yung link (halimbawa https://github.com/saints-script/saints-main-cheat) o ikaw ang gagawa ng GitHub account tapos ako yung magbibigay ng ready file para i-upload mo?

**Comment by nuryasinbakso408-png** [comment]


**Reply by shinybrightstar** [reply]
Hi @imtiazshawn – it looks like your post included a secret, token, or credential. To help protect your security, we’ve deleted the content.

You should be receiving this notice as an email notification. Please avoid sharing secrets or credentials in public spaces like GitHub Community Discussions going forward, as they could be misused.

For more information, please visit our docs on how to [Store Your Secrets Safely](https://docs.github.com/en/get-started/learning-to-code/storing-your-secrets-safely).

Thanks for helping us keep the community safe!

**Comment by MappleGaming208** [comment]
> ### Select Topic Area
> Product Feedback
> 
> ### Body
> I want to get a raw link like this "https://raw.githubusercontent.com/vsnaichuk/vsnaichuk/master/CV.pdf". I upload my resume to a github repository. but I don't get any raw option. how can I get it?


use ?raw=true so for ex i wanna make "https://github.com/MappleGaming208/Magic/blob/main/Scripts/lua/load.lua" into raw just do "https://github.com/MappleGaming208/Magic/blob/main/Scripts/lua/load.lua?raw=true"
it will redirect to "https://raw.githubusercontent.com/MappleGaming208/Magic/refs/heads/main/Scripts/lua/load.lua" youre welcome

**Comment by omprxkash** [comment]
Getting the "raw" link is simple. The raw link points directly to the plain text content of a file, without any GitHub website formatting (like the header, comments, or UI). This is super useful for downloading files directly or using them in scripts.

Here’s how you do it, step-by-step:

Navigate to the File: Go to the GitHub repository and click on the file you want the raw link for (e.g., a .txt, .js, .csv, or README.md file).

Find the "Raw" Button: On the top-right corner of the file's content box, you'll see several buttons. Look for the one that says Raw.

Click It!:
Click the Raw button. Your browser will now go to a new URL that looks something like this:
https://raw.githubusercontent.com/username/repository-name/branch-name/path/to/file.js

Copy the Link: That's it! You are now on the raw file page. Just copy the URL from your browser's address bar. This is your raw link.

**Reply by trnrzeek** [reply]
i am trying to do this with a webm video, but it just downloads the video over and over

**Reply by MappleGaming208Alt** [reply]
> i am trying to do this with a webm video, but it just downloads the video over and over

Pretty sure you can't put videos as raw files because in raw it's gonna show the code instead of media files.

**Comment by gabrielnascimentodeu-arch** [comment]
loadstring(game:HttpGet("https://raw.githubusercontent.com/Daivd977/Coquettekk/refs/heads/main/README.md"))()``

**Comment by Hanckdad** [comment]
![ca1b5f301a4ccae97b3a7fa12a18cc1a.jpg](https://github.com/user-attachments/assets/0d6a7519-5717-47d4-b832-c6c5356d0ea0)
